app/code/Magento/Ui/Component/Filters/Type/Input.php

```diff
@@ -67,14 +67,12 @@ protected function applyFilter()
         if (isset($this->filterData[$this->getName()])) {
             $value = str_replace(['%', '_'], ['\%', '\_'], $this->filterData[$this->getName()]);
 
-            if (!empty($value)) {
```

Comment:
Why not change to comparison with empty string instead?

Reply:
@orlangur Compared with the empty string.

Reply:
@jignesh-baldha thanks! Please change to strict comparison though. And pay attention to Stas suggestion please.

Reply:
`!== ''`